fuel-core/src/executor.rs

```diff
@@ -304,9 +318,22 @@ impl Executor {
                     }
                 }
                 Input::Contract { .. } => {}
-                // TODO: message validation waiting on db storage of messages + genesis
-                Input::MessageSigned { .. } => {}
-                Input::MessagePredicate { .. } => {}
+                Input::MessageSigned { message_id, .. }
+                | Input::MessagePredicate { message_id, .. } => {
+                    if let Some(message) = Storage::<MessageId, DaMessage>::get(db, message_id)? {
+                        if message.fuel_block_spend.is_some() {
+                            return Err(TransactionValidityError::MessageAlreadySpent(*message_id));
+                        }
+                        if BlockHeight::from(message.da_height) > block_da_height {
+                            return Err(TransactionValidityError::MessageSpendTooEarly(
+                                *message_id,
+                            ));
+                        }
+                    } else {
+                        println!("NOTFOUND {message_id:?}");
```

Comment:
We should use tracing instead of println's for logging messages. I'm suggesting we remove this message since the error should already have a trace/context.

```suggestion
```